src/zope/app/testing/tests.py

```diff
@@ -141,13 +141,44 @@
 class FunctionalHTTPDocTest(unittest.TestCase):
     maxDiff = None
 
+    assertRaisesRegex = getattr(unittest.TestCase, 'assertRaisesRegex',
+                                unittest.TestCase.assertRaisesRegexp)
+
     def test_dochttp(self):
         capture = NativeStringIO()
         dochttp(['-p', 'test', directory], output_fp=capture)
         got = capture.getvalue()
-
         self.assertEqual(expected, got)
 
+    def test_no_argument(self):
+        import sys
+        old_stderr = sys.stderr
+        sys.stderr = NativeStringIO()
+        try:
+            with self.assertRaises(SystemExit) as exc:
+                dochttp(["-p", 'test'])
+        finally:
+            sys.stderr = old_stderr
+
+        e = exc.exception
+        self.assertEqual(e.args, (2,))
+
+    def test_bad_directory_argument(self):
+        import tempfile
+        import shutil
+        d = tempfile.mkdtemp('.zope.app.testing')
+        self.addCleanup(shutil.rmtree, d)
+
+        with open(os.path.join(d, 'test1.request'), 'wt') as f:
+            f.write("Fake request file")
+        with open(os.path.join(d, 'test1.response'), 'wt') as f:
+            f.write("")
```

Comment:
I don't think I've ever seen `t` used in the open mode before.  Had to look it up! :)